lib/index.js

```diff
@@ -8,6 +8,9 @@
 const util = require("util");
 const memoize = require("./util/memoize");
 
+/** @typedef {import("../declarations/LoaderContext").EmptyContextAdditions} EmptyContextAdditions */
+/** @typedef {import("../declarations/LoaderContext").LoaderContext} LoaderContext */
+/** @typedef {import("../declarations/LoaderContext").LoaderDefinition} LoaderDefinition */
```

Comment:
```suggestion
/** @template T @typedef {import("../declarations/LoaderContext").LoaderDefinition<T>} LoaderDefinition<T> */
```

It's a bit weird to import generic types in jsdoc.

hmm... Maybe this will still not work as intended for the default value of the template argument.

If that doesn't work we may export two types: `LoaderDefinition` and `CustomLoaderDefintion<T>`...